experimental/tzif/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+# This file is part of ICU4X. For terms of use, please see the file
+# called LICENSE at the top level of the ICU4X source tree
+# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+[package]
+name = "icu_provider_tzif"
```

Comment:
Request: Moving forward, in general, all transformers / source providers should be defined within the `datagen` crate as a submodule under `transform`. The exception is if you make a general-purpose crate for parsing TZIF files, in which case you should name the crate something more general, and move it into the utils folder.

Reply:
I'd have utils/tzif and then stick the datagen stuff in experimental or under provider/datagen

but it's okay to use experimental now, up to you. experimental is anything goes :)

Reply:
Great. @sffc 

Do you have a preference on the ordering here? I'm finishing up the 2nd PR for the TZif parser itself right now. 

Would you prefer to move this PR over to `utils` or land in experimental and then move it all after it's reviewed and landed?

Reply:
I have no preference on ordering, whatever is most convenient for you. Shane might.

Reply:
If in doubt, keep it in `experimental`.

Reply:
Yeah, let's get all the code landed in `experimental` and then decide how to name it and where it should live.